### PR TITLE
Add Quill and Kinto into npm deps

### LINKED ISSUE #25: Add quill and kinto-http as npm dependencies?
This would make building and bundling a bit trickier (`npm install` and potentially moving files), but would keep some 3rd party code out of our codebase and make it easier to upgrade libraries in the future.

For example, it looks like we're bundling [**quill@1.2.4**](https://github.com/mozilla/notes/blob/master/src/sidebar/quill.js) in our repo, but the latest is [v1.2.6](https://www.npmjs.com/package/quill).

The minified [**kinto-http**](https://github.com/mozilla/notes/blob/master/src/sidebar/kinto-http.js) file doesn't even have a copyright header or version number I can find, so I don't even know what version of the library it is, and if it is out of date. Looks like the latest version on npm is [v4.3.4](https://www.npmjs.com/package/kinto-http) though.

### PR DESCRIPTION
**Pull request by vladikoff**
Fixes #25

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -1,4 +1,4 @@
 src/sidebar/kinto-http.js
 src/sidebar/quill.js
```
Comment (pdehaan) [inline .eslintignore]: Not sure if these lines are needed anymore now that you've deleted from repo and moved into "src/sidebar/vendor/*".

File: scripts/postinstall.sh [hunk 2]
```diff
@@ -3,3 +3,6 @@
 set -e
 
 cp node_modules/testpilot-metrics/testpilot-metrics.js src/vendor
+cp node_modules/kinto-http/dist/kinto-http.js src/sidebar/vendor
+cp node_modules/quill/dist/quill.js src/sidebar/vendor
```
Comment (pdehaan) [inline scripts/postinstall.sh]: Not sure about your feelings on using _quill.js (409K)_ vs _quill.**min**.js (202K)_. 🤷‍♀️ 

```sh
➜  notes git:(master) ✗ ls -lash node_modules/quill/dist

  56 -rw-r--r--   1 pdehaan  staff    24K Jun  5 10:21 quill.bubble.css
  24 -rw-r--r--   1 pdehaan  staff   8.6K Jun  5 10:21 quill.core.css
 568 -rw-r--r--   1 pdehaan  staff   283K Jun  5 10:21 quill.core.js
 824 -rw-r--r--   1 pdehaan  staff   409K Jun  5 10:21 quill.js
 408 -rw-r--r--   1 pdehaan  staff   202K Jun  5 10:21 quill.min.js
1904 -rw-r--r--   1 pdehaan  staff   951K Jun  5 10:21 quill.min.js.map
  48 -rw-r--r--   1 pdehaan  staff    24K Jun  5 10:21 quill.snow.css
```

File: scripts/postinstall.sh [hunk 3]
```diff
@@ -3,3 +3,6 @@
 set -e
 
 cp node_modules/testpilot-metrics/testpilot-metrics.js src/vendor
+cp node_modules/kinto-http/dist/kinto-http.js src/sidebar/vendor
```
Comment (pdehaan) [inline scripts/postinstall.sh]: Do you want the non-minimized _kinto-http.js (1.0M)_ vs _kinto-http.**min**.js (230K)_?

```sh
➜  notes git:(master) ✗ ls -lash node_modules/kinto-http/dist

2024 -rw-r--r--   1 pdehaan  staff   1.0M May  9 15:29 kinto-http.js
 464 -rw-r--r--   1 pdehaan  staff   230K May  9 15:29 kinto-http.min.js
 144 -rw-r--r--   1 pdehaan  staff    70K May  9 15:29 kinto-http.noshim.js
 200 -rw-r--r--   1 pdehaan  staff    99K May  9 15:29 moz-kinto-http-client.js
```

File: .eslintignore (hunk 1 above)
Comment (vladikoff) [inline .eslintignore]: Yeap we can remove that

**Pull request merged**